PSKoans/koanOrder.md

```diff
@@ -0,0 +1,50 @@
+# PowerShell Koan Order
```

Comment:
This is a good idea, but if we're gonna do this in Markdown we should probably add links to each entry pointing at the file in the repo :)

Reply:
Also, I'd preemptively add a `Module` column for future-proofing.

Reply:
I've added it but it contradicts what I feel to be the primary use case for this file; aiding a developer in understanding the overall structure of the course and thus informing them where they can place their Koan, or if they need to amend other Koans for theirs to fit in.

Such an activity I'd imagine would take place in an IDE, such as Visual Studio Code. And at least in my Visual Studio Code the markdown file is not user friendly with the added links.

I'm double checking all the links now so maybe expect another few commits until they're all working as they should 😄

Reply:
All links working.

Also here's a thought... maybe add a step to the Azure Pipeline to automatically create/update the markdown table? I'm presuming there's already a cmdlet that produces an array of all the koans, their number and their location, in which case you could just iterate through that to generate the markdown?

Prevents documentation getting out of date and removes admin overhead.

Reply:
This isn't a bad idea, but I'm hesitant to add any pipeline steps that actually push files back to the repo. 🤔 

Definitely something to consider, but we'll think more on that later, I reckon. 😄

Reply:
Fair enough 😄 

Last conference I went to they covered making your own custom tasks for Azure Pipelines in either JavaScript or PowerShell... so it's something I could look at in the future 🦄